### PR TITLE
feat: Support symbolic expressions in `from_ir`

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

wrap symbols from the local simulator as free parameter expressions

*Testing done:*

unit and manual

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/circuits/braket_program_context.py [hunk 1]
```diff
@@ -10,10 +10,11 @@
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
-
+from functools import singledispatchmethod
```
Comment (speller26) [inline src/braket/circuits/braket_program_context.py:13]:
```suggestion

from functools import singledispatchmethod
```

File: src/braket/circuits/braket_program_context.py [hunk 2]
```diff
@@ -139,3 +141,11 @@ def add_result(self, result: Results) -> None:
...
         """
         self._circuit.add_result_type(braket_result_to_result_type(result))
+
+    @singledispatchmethod
+    def handle_parameter_value(self, value):
+        return value
+
+    @handle_parameter_value.register
+    def _(self, value: Expr):
+        return FreeParameterExpression(str(value))
```
Comment (speller26) [inline src/braket/circuits/braket_program_context.py:151]: Use a private method for single dispatch; otherwise, the docs start looking really ugly

Comment (ajberdy) [reply src/braket/circuits/braket_program_context.py:151]: Good call

Comment (speller26) [reply src/braket/circuits/braket_program_context.py:151]:
```suggestion
    def handle_parameter_value(self, value):
        return self._handle_parameter_value(value)

    @singledispatchmethod
    def _handle_parameter_value(self, value):
        return value

    @_handle_parameter_value.register
    def _(self, value: Expr):
        return FreeParameterExpression(str(value))
```

**Review by speller26 — APPROVED** [review]

**Pull request merged**